services/ui_backend_service/api/run.py

```diff
@@ -104,10 +104,16 @@ async def get_all_runs(self, request):
         builtin_conditions, _ = builtin_conditions_query(request)
         has_tag_filter = len([s for s in builtin_conditions if 'tags||system_tags' in s]) > 0
         if has_tag_filter:
-            _, _, _, order, _, _ = pagination_query(request, allowed_order=allowed_order, allowed_group=[])
-            if len(order) > 0:
+            allowed_optimized_order = ['flow_id', 'ts_epoch']
+            allowed_unoptimized_order = [o for o in allowed_group if o not in ['flow_id', 'ts_epoch']]
```

Comment:
could instead be `if o not in allowed_optimized_order` in case the above gets changed?